lib/chef/provider/package/aix.rb

```diff
@@ -108,28 +114,63 @@ def candidate_version
         # So far, the code has been tested only with standalone packages.
         #
         def install_package(name, version)
-          Chef::Log.debug("#{@new_resource} package install options: #{@new_resource.options}")
-          if @new_resource.options.nil?
-            shell_out_with_timeout!( "installp -aYF -d #{@new_resource.source} #{@new_resource.package_name}" )
-            Chef::Log.debug("#{@new_resource} installed version #{@new_resource.version} from: #{@new_resource.source}")
+          Chef::Log.debug("#{new_resource} package install options: #{new_resource.options}")
+          if new_resource.options.nil?
+            shell_out_with_timeout!( "installp -aYF -d #{new_resource.source} #{new_resource.package_name}" )
+            Chef::Log.debug("#{new_resource} installed version #{new_resource.version} from: #{new_resource.source}")
           else
-            shell_out_with_timeout!( "installp -aYF #{expand_options(@new_resource.options)} -d #{@new_resource.source} #{@new_resource.package_name}" )
-            Chef::Log.debug("#{@new_resource} installed version #{@new_resource.version} from: #{@new_resource.source}")
+            shell_out_with_timeout!( "installp -aYF #{expand_options(new_resource.options)} -d #{new_resource.source} #{new_resource.package_name}" )
+            Chef::Log.debug("#{new_resource} installed version #{new_resource.version} from: #{new_resource.source}")
           end
         end
 
         alias_method :upgrade_package, :install_package
 
         def remove_package(name, version)
-          if @new_resource.options.nil?
+          if new_resource.options.nil?
             shell_out_with_timeout!( "installp -u #{name}" )
-            Chef::Log.debug("#{@new_resource} removed version #{@new_resource.version}")
+            Chef::Log.debug("#{new_resource} removed version #{new_resource.version}")
           else
-            shell_out_with_timeout!( "installp -u #{expand_options(@new_resource.options)} #{name}" )
-            Chef::Log.debug("#{@new_resource} removed version #{@new_resource.version}")
+            shell_out_with_timeout!( "installp -u #{expand_options(new_resource.options)} #{name}" )
+            Chef::Log.debug("#{new_resource} removed version #{new_resource.version}")
           end
         end
 
+        def filter_source_device
+          return unless new_resource.source
+
+          file_sets = []
+          ret = shell_out_with_timeout("installp -L -d #{new_resource.source}")
+          unless ret.exitstatus == 0
+            raise Chef::Exceptions::Package, "installp -L -d #{new_resource.source} - #{ret.format_for_exception}!"
+          end
+          ret.stdout.each_line do |line|
+            case line
+            when /(^|:)#{new_resource.package_name}:/
```

Comment:
I have a personal preference for including an example of a line that we're matching as a comment above a regex like this. It helps make your intentions more clear to the next person.